### PR TITLE
OB-996: add document about new asset code

### PR DESCRIPTION
**Pull request by CaillaudPA**
Provide a sample into our documentation to link assets from supplier to products with product link rule.

[JIRA](https://akeneo.atlassian.net/browse/OB-996)

### CODE REVIEW

File: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md [hunk 1]
```diff
@@ -24,10 +24,25 @@ Suppliers can only share assets belonging to an asset family if their main media
 If the attribute used as the main media of the asset family linked to this asset collection is a media link attribute, the collection won't be synchronized on the supplier's end and they won't be able to send the assets.
 :::
 
-## Asset code uniqueness
+##  New asset code uniqueness
+
+Before, asset code was generating using only UUID, now there is a new asset code generation
```
Comment (phaseinducer) [inline content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:29]:
```suggestion
Before, asset code was generated using only UUID, now there is a new asset code generation.
```

Comment (phaseinducer) [reply content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:29]: Can you be more specific? Is it the generation you explain below?

File: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md [hunk 2]
```diff
@@ -24,10 +24,25 @@ Suppliers can only share assets belonging to an asset family if their main media
...
+
 As many suppliers can be working on the same asset collection at the same time, they could set the same asset code twice. The Onboarder puts a constraint on suppliers to guarantee the uniqueness of asset codes among all suppliers.
 * A supplier can't create an asset code
-* The asset code is automatically generated using a UUID
+* The asset code is automatically generated using a concatenation of :
```
Comment (phaseinducer) [inline content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:33]:
```suggestion
* The asset code is automatically generated using a concatenation of:
```

File: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md [hunk 3]
```diff
@@ -24,10 +24,25 @@ Suppliers can only share assets belonging to an asset family if their main media
...
+  - supplier code
+  - filename
+  - uuid
```
Comment (phaseinducer) [inline content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:36]:
```suggestion
  - UUID
```

File: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md [hunk 4]
```diff
@@ -24,10 +24,25 @@ Suppliers can only share assets belonging to an asset family if their main media
...
+
+For example, the supplier 'pac' upload new asset called brown-chair.jpg, the asset code will be something like :
```
Comment (phaseinducer) [inline content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:38]:
```suggestion
For example, the supplier 'pac' uploads a new asset called brown-chair.jpg, the asset code will be something like:
```

File: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md [hunk 5]
```diff
@@ -24,10 +24,25 @@ Suppliers can only share assets belonging to an asset family if their main media
...
+ - pac_brown_chair_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg
+
+You can easily create a rule on it, you can guess from the beginning of the asset code, which supplier added it
+
+Regex could be : ```/^pac_/``` meaning the asset code must start by this string
```
Comment (phaseinducer) [inline content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:43]:
```suggestion
Regex could be : ```/^pac_/```, meaning the asset code must start with this string.
```

File: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md [hunk 6]
```diff
@@ -24,10 +24,25 @@ Suppliers can only share assets belonging to an asset family if their main media
...
-* The asset code is automatically generated using a UUID
+* The asset code is automatically generated using a concatenation of :
+  - supplier code
+  - filename
+  - uuid
+
+For example, the supplier 'pac' upload new asset called brown-chair.jpg, the asset code will be something like :
+ - pac_brown_chair_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg
+
+You can easily create a rule on it, you can guess from the beginning of the asset code, which supplier added it
```
Comment (phaseinducer) [inline content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:41]: Not clear for me :/

File: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md [hunk 7]
```diff
@@ -24,10 +24,25 @@ Suppliers can only share assets belonging to an asset family if their main media
...
+  - supplier code
+  - filename
+  - UUID
+
+For example, the supplier 'pac' uploads new asset called brown-chair.jpg, the asset code will be something like:
+ - pac_brown_chair_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg
+
+You can easily create a rule on it, you can guess from the beginning of the asset code, which supplier added it
+
+Regex could be : ```/^pac_/```, meaning the asset code must start with this string
```
Comment (PierrickMartos) [inline content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:43]: Thank you for this PR :) 

The multiline commit suggestion didn't work when there is line deletion, so here is the changes below and my comments here:
- The example isn't enough clear enough for a user to setup a product link rule
- Actually, the asset code format isn't present

```
## Asset code format and uniqueness

As many suppliers can be working on the same asset collection at the same time, they could have the same asset code twice. The Onboarder puts a constraint on suppliers to guarantee the uniqueness of asset codes among all suppliers.
* A supplier can't create an asset code
* The asset code is automatically generated by replacing hyphens with underscores and by using a concatenation of:
  - supplier code
  - filename
  - UUID

Here is the asset code format: `<supplier_code>_<filename>_<uuid>`

For example, the supplier `pac` upload a new asset called `chair-packshot.jpg`, the asset code will be: `pac_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`

You can easily create product link rules to automatically link assets to products/product models based on assets filenames (More info on [https://help.akeneo.com/pim/serenity/articles/assets-product-link-rules.html](product link rules)).
On our example, regex could be: `(?<supplier_code>.*?)_(?<product_ref>.*?)_(?<attribute_ref>.*?)_(.*).jpg`

::: warning
Previous uploaded assets will still have the old asset code format, you need to reupload them to have the new format.
::::
```

**Review by ValentinMerlet — APPROVED** [review]

File: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md [hunk 8]
```diff
@@ -24,10 +24,25 @@ Suppliers can only share assets belonging to an asset family if their main media
...
 * A supplier can't create an asset code
-* The asset code is automatically generated using a UUID
+* The asset code is automatically generated by replacing hyphens with underscores and by using a concatenation of:
+  - supplier code
+  - filename
+  - UUID
+
+Here is the asset code format: `<supplier_code>_<filename>_<uuid>`
+
+For example, the supplier `pac` upload a new asset called `chair-packshot.jpg`, the asset code will be: `pac_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`
```
[the remark this answers is not in the document]
Comment (PierrickMartos) [reply content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:38]:
```suggestion
For example, the supplier `pac` uploads a new asset called `chair-packshot.jpg`, the asset code will be: `pac_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`
```

**Review by alexandrelissowski — APPROVED** [review]
Just a single comment on formating the hyperlink. Yohan and Pierrick already review the doc. It is very clear. Thanks a lot

File: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md [hunk 9]
```diff
@@ -24,10 +24,25 @@ Suppliers can only share assets belonging to an asset family if their main media
...
 * A supplier can't create an asset code
-* The asset code is automatically generated using a UUID
+* The asset code is automatically generated by replacing hyphens with underscores and by using a concatenation of:
+  - supplier code
+  - filename
+  - UUID
+
+Here is the asset code format: `<supplier_code>_<filename>_<uuid>`
+
+For example, the supplier `supplier1` upload a new asset called `chair-packshot.jpg`, the asset code will be: `supplier1_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`
```
Comment (PierrickMartos) [inline content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md:38]:
```suggestion
For example, the supplier `supplier1` uploads a new asset called `chair-packshot.jpg`, the asset code will be: `supplier1_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`
```

**Pull request merged**